src/param.c

```diff
@@ -142,6 +142,63 @@ stumpless_new_param( const char *name, const char *value ) {
   return result;
 }
 
+struct stumpless_param *
+stumpless_new_param_from_string( const char *string ) {
+
+  VALIDATE_ARG_NOT_NULL( string );
+
+  int i;
+  int name_len = 0, name_start = 0, value_len = 0, value_start = 0;
```

Comment:
These should be `size_t`, as they are all used as sizes/indexes, and should not be interpreted as signed in any case.